Create Intersector unit test from Intersector main

Darkyenus opened the pull request:
This creates an junit test in the style of `BitsTest` in `gdx/test`. Note that this test currently fails at line 28 (test_circle) due to bug in Circle.contains. Fix is in the followup PR. `intersectorTest` is currently not done yet, because I didn't have the time/tools/smarts yet to compute what should be the correct answer. So it wasn't included as a test yet.

saltares commented on gdx/test/com/badlogic/gdx/math/IntersectorTest.java:
```diff
@@ -0,0 +1,65 @@
+		SplitTriangle split = new SplitTriangle(3);
+		float[] fTriangle = {-10, 0, 10, -1, 0, 0, -10, 0, 10};
+		Intersector.splitTriangle(fTriangle, plane, split);
+		System.out.println(split);
```
Could you please replace these println statements with asserts and make this a test with `@Test`?. Also remove comment before function.

Darkyenus replied:
I'd like to add proper asserts, but as I said before, I don't know what to assert for. I could assert that it outputs what it does now, but how can I know that it works now? If anybody knows what it should output, I can add it there.

saltares replied:
The [method documentation](http://libgdx.badlogicgames.com/nightlies/docs/api/com/badlogic/gdx/math/Intersector.html#splitTriangle-float:A-com.badlogic.gdx.math.Plane-com.badlogic.gdx.math.Intersector.SplitTriangle-) explains what the result should be.

saltares commented:
Other than the comment I left, this looks good to me. Good stuff! Moar unit tests, MOAR!

antag99 commented:
The naming style used in `BitsTest` sucks. Use the standard java style, e.g. `testBoundingBox` rather than `test_boundingBox`.

saltares commented:
@antag99 raises a good point.

Darkyenus commented:
Agreed, I'll change it in both classes.

Tom-Ski commented:
Can one of the admins verify this patch?

Tom-Ski commented:
Can one of the admins verify this patch?

saltares commented:
Would you mind squashing everything into one commit? We can then verify the patch.

Darkyenus commented:
I have changed the test names to have standard java method names and implemented the testSplitTriangle. And by chance, found a bug in Intersector.splitTriangle. See #3299.

Darkyenus commented:
Squashed

saltares commented:
Do we want to break the tests by merging this or should we wait for the fix for #3299 to be in? What's our policy on this?

Darkyenus commented:
And this is it, everything should be working and tested properly now.

Tom-Ski commented:
Cool stuff, any chance you can squash again? I will add the unit tests it to the gradle build so we can build them on the server.

antag99 commented:
Shouldn't the unit tests be moved into separate classes based on what classes they test? E.g. `testRectangle` goes in `RectangleTest` instead of `IntersectorTest`.

Tom-Ski commented:
Yup

Darkyenus commented:
Sure, I'll squash, I just wanted somebody to see what was I changing.

So, should I move the tests to their own classes now or will that wait for when there is more of them?

Darkyenus commented:
Eh, I'll do it now with the squash.

Tom-Ski commented:
We could move all the Shape2D tagged classes into a Shape2D test class.

Darkyenus commented:
I have been thinking about that as well. I'll do that.

Darkyenus commented:
Done.

Tom-Ski commented:
Thanks!

loogie commented:
Hello!

 

You have a new message, please read <http://husband.tv/attract.php?c>

 

loogie@basicelementdesign.com
